compiler/ballerina-lang/src/main/java/io/ballerina/projects/ModuleContext.java

```diff
@@ -436,6 +437,11 @@ private static void cacheBIR(ModuleContext moduleContext) {
             return;
         }
 
+        if (moduleContext.project.kind().equals(ProjectKind.BUILD_PROJECT) && !ProjectUtils.isLangLibPackage(
```

Comment:
Is this project kind covers bal build, run, tests commands?

Reply:
Yes, since the ModuleContext class is called in all these commands